src/main/scala/com/vngrs/etl/Combine.scala

```diff
@@ -0,0 +1,49 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * [[com.vngrs.etl.Combine]] is an interface for combining two [[com.vngrs.etl.Pipeline]]s into one.
+  *
+  * @tparam A Type of Input [[com.vngrs.etl.Pipeline]]
+  * @tparam B Type of Output [[com.vngrs.etl.Pipeline]]
+  */
+trait Combine[A, B] {
```

Comment:
I agree understanding combiner handles two pipelines is almost impossible nevertheless we have a factory method which explains it a little bit and handles it. I feel like we are consuming the time we don't have currently. If all of you guys thinks that it should be removed then I will. However, if it is not decided yet then it should be removed in another PR for the sake of the project.